homeassistant/components/mqtt/light/schema_basic.py

```diff
@@ -689,31 +713,41 @@ async def publish(topic, payload):
                 self._config[CONF_ENCODING],
             )
 
-        def scale_rgbx(color, brightness=None):
+        def scale_rgbx(
+            color: tuple[int, ...],
+            brightness: int | None = None,
+        ) -> tuple[int, ...]:
             """Scale RGBx for brightness."""
             if brightness is None:
                 # If there's a brightness topic set, we don't want to scale the RGBx
                 # values given using the brightness.
                 if self._topic[CONF_BRIGHTNESS_COMMAND_TOPIC] is not None:
                     brightness = 255
                 else:
-                    brightness = kwargs.get(ATTR_BRIGHTNESS, self.brightness or 255)
+                    brightness = kwargs.get(ATTR_BRIGHTNESS) or self.brightness or 255
             return tuple(int(channel * brightness / 255) for channel in color)
 
-        def render_rgbx(color, template, color_mode):
+        def render_rgbx(
+            color: tuple[int, ...],
+            template: str,
+            color_mode: ColorMode,
+        ) -> PublishPayloadType:
             """Render RGBx payload."""
-            if tpl := self._command_templates[template]:
-                keys = ["red", "green", "blue"]
-                if color_mode == ColorMode.RGBW:
-                    keys.append("white")
-                elif color_mode == ColorMode.RGBWW:
-                    keys.extend(["cold_white", "warm_white"])
-                rgb_color_str = tpl(variables=zip(keys, color))
-            else:
-                rgb_color_str = ",".join(str(channel) for channel in color)
-            return rgb_color_str
-
-        def set_optimistic(attribute, value, color_mode=None, condition_attribute=None):
+            rgb_color_str = ",".join(str(channel) for channel in color)
+            keys = ["red", "green", "blue"]
+            if color_mode == ColorMode.RGBW:
+                keys.append("white")
+            elif color_mode == ColorMode.RGBWW:
+                keys.extend(["cold_white", "warm_white"])
+            variables = cast(TemplateVarsType, zip(keys, color))
+            return self._command_templates[template](rgb_color_str, variables)
```

Comment:
I think you can just pass `None` here (and remove `rgb_color_str`):
```suggestion
            return self._command_templates[template](None, variables)
```

Reply:
If no template is set the content of `rgb_color_str` will be the output.